### PR TITLE
Replace None with Py_None and some other fixing.

### PR DESCRIPTION
**Pull request by Daetalus**
Spit it from tk_support PR. I will improve that PR on weekend.

### CODE REVIEW

File: src/runtime/builtin_modules/thread.cpp [hunk 1]
```diff
@@ -227,6 +233,9 @@ void setupThread() {
                                              FunctionMetadata::create((void*)getIdent, BOXED_INT, 0), "get_ident"));
     thread_module->giveAttr("stack_size", new BoxedBuiltinFunctionOrMethod(
                                               FunctionMetadata::create((void*)stackSize, BOXED_INT, 0), "stack_size"));
+    thread_module->giveAttr(
```
Comment (undingen) [inline src/runtime/builtin_modules/thread.cpp]: this needs a test

File: src/runtime/builtin_modules/sys.cpp [hunk 2]
```diff
@@ -816,6 +816,8 @@ void setupSys() {
     sys_flags_cls->giveAttr(
         "__new__", new BoxedFunction(FunctionMetadata::create((void*)BoxedSysFlags::__new__, UNKNOWN, 1, true, true)));
     sys_flags_cls->tp_dealloc = (destructor)BoxedSysFlags::dealloc;
+
+    sys_flags_cls->giveAttr("ignore_environment", boxInt(Py_IgnoreEnvironmentFlag));
```
Comment (undingen) [inline src/runtime/builtin_modules/sys.cpp]: I don't like this half implemented `sys.flags` thing but I don't think you change makes it worse so I'm fine with merging this but I created https://github.com/dropbox/pyston/issues/1304 to make sure we fix it at some point

**Comment by Daetalus** [comment]
Working on scikit-learn support. So move the `thread.interrupt_main` to `tk_support` PR. Will take a look on weekend.

**Comment by undingen** [comment]
thanks for the change! :+1:

**Pull request merged**